File: water-system/src/main/java/com/water/water_system/entity/Tenant.java
package com.water.water_system.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.List;
import java.util.ArrayList;
import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@Table(name = "tenants")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Tenant {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private String name;
    
    @Column(unique = true)
    private String phone;
    
    @Enumerated(EnumType.STRING)
    private Role role = Role.TENANT;

    private Integer previousUnits = 0;
    private String doorNumber;

    @OneToMany(mappedBy = "tenant", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<Unit> units = new ArrayList<>();

    @OneToMany(mappedBy = "tenant", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<Payment> payments = new ArrayList<>();
    
    public enum Role {
        ADMIN,
        TENANT
    }
    
    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; }
    public Role getRole() { return role; }
    public void setRole(Role role) { this.role = role; }
    public Integer getPreviousUnits() { return previousUnits; }
    public void setPreviousUnits(Integer previousUnits) { this.previousUnits = previousUnits; }
    public String getDoorNumber() { return doorNumber; }
    public void setDoorNumber(String doorNumber) { this.doorNumber = doorNumber; }
}

File: water-backend/app/Models/Tenant.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Tenant extends Model
{
    protected $fillable = [
        'name',
        'phone',
        'role',
        'previousUnits',
        'doorNumber'
    ];

    public $timestamps = false;

    // Role constants
    public const ROLE_ADMIN = 'ADMIN';
    public const ROLE_TENANT = 'TENANT';

    public function units(): HasMany
    {
        return $this->hasMany(Unit::class);
    }

    public function payments(): HasMany
    {
        return $this->hasMany(Payment::class);
    }
}
